platform/onebox/backup_and_restore/registry/export.sh

```diff
@@ -0,0 +1,146 @@
+    exit 1
+  fi
+
+  echo "$green $(date) Successfully validated user input $default"
+}
+
+function get_image_list() {
+	# sqlcmd -S tcp:${DB_CONN} -d ${DB_NAME} -U ${DB_USER} -P ${DB_PASSWORD} -i getimages.sql -o images.txt -h -1 -W
+	sqlcmd -S tcp:${DB_CONN} -d ${DB_NAME} -U ${DB_USER} -P ${DB_PASSWORD} -o images.txt -h -1 -W -Q "set nocount on; select distinct mpi.image_uri from ml_package_images mpi inner join ml_skill_versions msv on msv.ml_package_version_id = mpi.version_id and  msv.processor = mpi.processor where msv.status in ('UPDATING', 'COMPLETED', 'VALIDATING_DEPLOYMENT') and mpi.status = 'ACTIVE'"
+}
+
+formulate_docker_command() {
+    docker images registry
+    if [[ $? -ne 0 ]]; then
+        echo "sudo permission required for docker"
+        DOCKER_COMMAND="sudo docker"
+    else
+        DOCKER_COMMAND="docker"
+    fi
+}
+
+function docker_setup() {
+	# Mark docker registry as unauth => Adapt for other envs
+	echo "{\"insecure-registries\": [\"${REGISTRY_ENDPOINT}\"]}" > insecure.json
+	sudo touch /etc/docker/daemon.json
+	daemondiff=$(sudo jq -s '.[0] as $o1 | .[1] as $o2 | ($o1 + $o2) | ."insecure-registries" = ($o1."insecure-registries" + $o2."insecure-registries" | unique)'  /etc/docker/daemon.json insecure.json)
+	sudo echo $daemondiff > /etc/docker/daemon.json
+	# Restart docker
+	sudo service docker restart
+	# Login to docker registry
+	${DOCKER_COMMAND} login ${REGISTRY_ENDPOINT} -u ${REGISTRY_USER} -p ${REGISTRY_PASSWORD}
+}
+
+function save_image() {
+	${DOCKER_COMMAND} pull $1
+	if [ $? -ne 0 ];
+	then
+		echo "$red $(date) Not able to pull image for $1 $default"
+		exit 1
+	fi
+	set -o pipefail
+	${DOCKER_COMMAND} save $1 | gzip > $2
+	if [ $? -ne 0 ];
+	then
+		echo "$red $(date) Not able to save image for $1 in $2 $default"
+		rm -rf $2
+		exit 1
+	fi
+}
+
+function save_images() {
+	mkdir -p ${EXPORT_PATH}
+    # loop over images, replace existing registry tag with the one from nodeport and generate tars if they don't exist in destination
+    # Pass full image and a file name to save_image
+    while read img; do
+  		# each line will be json
+  		echo "$img"
+  		
+  		registry=$(echo $img | jq -r 'select(.repository != null) | .repository')
+  		image=$(echo $img | jq -r 'select(.imageName != null) | .imageName')
+  		tag=$(echo $img | jq -r 'select(.imageTag != null) | .imageTag')
+
+  		if [[ $registry == ${OLD_REGISTRY_ENDPOINT}* ]]; then
+  			newimg=${REGISTRY_ENDPOINT}/${image}:${tag}
+  			imgName=$(echo ${newimg##*/})
+  			if [ ! -f "${EXPORT_PATH}/$imgName.tar.gz" ]; then
+    			echo "$green $(date) ${EXPORT_PATH}/$imgName file does not exist, Generating image tar $default"
+    			save_image $newimg ${EXPORT_PATH}/${imgName}.tar.gz
+    		else
+    			echo "$green $(date) ${EXPORT_PATH}/$imgName file exists, Skipping $default"
+  			fi
+  		fi
+	done <images.txt
+}
+
+validate_setup
+
+validate_input
+
+docker_setup
+
+get_image_list
+
+save_images
```

Comment:
at the start of the script, check if sufficient space is available on the disk to export the images and also recommend users to use a secondary disk which is non root disk to avoid kubernetes getting impacted because of disk space issue

Reply:
@rajivml the script is run on a separate VM. Hence not valid

Reply:
yeah but the path where they are exporting should have at least 50 -100 gigs of free space to export the same otherwise the export might fail in the middle right